presto-oracle/src/test/java/io/prestosql/plugin/oracle/TestOracleTypes.java

```diff
@@ -0,0 +1,185 @@
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestOracleTypes
+        extends AbstractTestQueryFramework
+{
+    private OracleServer oracleServer;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        this.oracleServer = new OracleServer();
+        return createOracleQueryRunner(oracleServer, ImmutableMap.of());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public final void destroy()
+    {
+        if (oracleServer != null) {
+            oracleServer.close();
+        }
+    }
+
+    @Test
+    public void testVarcharType()
+    {
+        new UnderTest()
+                .withType("varchar", "(10)")
+                .withValues("test")
+                .doTest();
+        new UnderTest()
+                .withType("varchar", null, "(4000)")
+                .withValues("test")
+                .doTest();
+    }
+
+    @Test
+    public void testNumericTypes()
+    {
+        new UnderTest()
+                .withType("tinyint")
+                .withValues(123, (byte) 123)
+                .doTest();
+        new UnderTest()
+                .withType("smallint")
+                .withValues(123, (short) 123)
+                .doTest();
+        new UnderTest()
+                .withType("integer")
+                .withValues(123, 123)
+                .doTest();
+        new UnderTest()
+                .withType("bigint")
+                .withValues(123, 123L)
+                .doTest();
+        new UnderTest()
+                .withType("decimal", null, "(38,0)")
+                .withValues(123, BigDecimal.valueOf(123))
+                .doTest();
+        new UnderTest()
+                .withType("decimal", "(5,1)")
+                .withValues(123, BigDecimal.valueOf(123).setScale(1, RoundingMode.HALF_UP))
+                .doTest();
+        new UnderTest()
+                .withType("decimal", "(5,2)")
+                .withValues(123, BigDecimal.valueOf(123).setScale(2, RoundingMode.HALF_UP))
+                .doTest();
+        new UnderTest()
+                .withType("decimal", "(5,2)")
+                .withValues(123.046, BigDecimal.valueOf(123.05).setScale(2, RoundingMode.HALF_UP))
+                .doTest();
+    }
+
+    @Test
+    public void testClobColumn()
+    {
+        oracleServer.execute("CREATE TABLE test_clob_column (col CLOB)");
+        assertUpdate("INSERT INTO test_clob_column (col) VALUES ('clob')", 1);
+
+        assertTrue(getQueryRunner().tableExists(getSession(), "test_clob_column"));
+        assertQuerySucceeds("select count(1) from test_clob_column");
+        assertQuery("SELECT * FROM test_clob_column", "VALUES ('clob')");
+
+        assertUpdate("DROP TABLE test_clob_column");
+        assertFalse(getQueryRunner().tableExists(getSession(), "test_clob_column"));
+    }
+
+    private class UnderTest
```

Comment:
It looks like `DataTypeTest` concept, but separate.
What do we need to change in `DataTypeTest` to avoid copying the mechanism here?

Reply:
I think we have to add expected type and value
for example:
        DataTypeTest.create()
                .addRoundTrip(varcharDataType(), "test") // this will fail because its tranforme to varchar(4000)
                .addRoundTrip(varcharDataType(5000), "test") // // this will fail because its tranforme to varchar, clob in oracle
                .execute(getQueryRunner(), prestoCreateAsSelect("test_basic_types"));

Reply:
changed to DataTypeTest, I have to add one method to work